frontend/app/src/scenes/Home/scenes/AddSubCategory/index.tsx

```diff
@@ -54,25 +60,65 @@ class AddSubCategory extends React.Component<
       );
   };
 
-  onFinish = (values: any) => {
-    this.setState({ isLoading: true });
-    const subCategory = {
-      name: values.name,
-      category: {
-        id: values.category,
-      },
-    };
+  fetchSubCategoryDetails = (subCategoryId: string) => {
     axios
-      .post(
+      .get(
+        window.location.origin +
+          `/core/academix/sub-categories/${subCategoryId}`
+      )
+      .then((result: AxiosResponse<SubCategory>) => {
+        if (result.status == 200) {
+          this.setState({ subcategory: result.data, isLoading: false });
+        } else {
+          throw new Error();
+        }
+      })
+      .catch((error) =>
+        handleApiError(
+          error,
+          'Something went wrong when trying to load the subcategory'
+        )
+      );
+  };
+
+  onFinish = (values: any) => {
+    let statusCode: number;
+    let method;
+    if (this.subCategoryId == null) {
+      statusCode = 201;
+      const subCategory = {
+        name: values.name,
+        category: {
+          id: values.category,
+        },
+      };
+      method = axios.post(
         window.location.origin + '/core/academix/admin/sub-categories/',
         subCategory
-      )
-      .then((res: AxiosResponse<SubCategory>) => {
-        if (res.status == 201) {
+      );
+    } else {
+      statusCode = 202;
+      const subCategory = {
+        id: this.subCategoryId,
+        name: values.name,
+        category: {
+          id: values.category,
+        },
+      };
+      method = axios.put(
+        window.location.origin +
+          `/core/academix/admin/sub-categories/${this.subCategoryId}`,
+        subCategory
+      );
+    }
+    this.setState({ isLoading: true });
+    method
+      .then((result: AxiosResponse<SubCategory>) => {
+        if (result.status == statusCode) {
           this.setState({ isLoading: false });
           notification.success({
             message: 'Success!',
-            description: 'Successfully Created a subcategory',
+            description: 'The category saved successfully!',
```

Comment:
Shouldn't this be `The subcategory saved successfully!` ?

Reply:
yeah it should be `subcategory` my mistake. @Gravewalker666